rails-commenting.rb

```diff
@@ -7,45 +7,45 @@
 # FILE: app/controller/blog_posts_controller.rb
 
 # ---1)
-class BlogPostsController < ApplicationController
+class BlogPostsController < ApplicationController # Creates a controller class for BlogPosts that inherits all behavior from ApplicationController
   def index
     # ---2)
-    @posts = BlogPost.all
+    @posts = BlogPost.all # Selects every entry from the BlogPost table and assigns it to the instance variable @posts. Returns this instance variable as it is the only line of code in this method
   end
 
   def show
     # ---3)
-    @post = BlogPost.find(params[:id])
+    @post = BlogPost.find(params[:id]) # Returns an instance variable containing a single blog post located by the id passed into the request params
   end
 
   # ---4)
-  def new
+  def new # Defines a method that creates an instance variable of an empty Post object built from the Post class. This entry is not saved to the database
```

Comment:
I can see from the *edit* that you're aware that these associate with views-- calling it good.